wavelength-calibrate_new.py

```diff
@@ -49,39 +50,72 @@
 
 coeffj = np.dot(np.linalg.pinv(coj),wavj); # the model
 
+coj1 = np.zeros((len(wavj1),16))
+
+for k in range(len(wavj1)-1):
+    c = 0
+    for i in range(3):
+        for j in range(3):
+        
+            coj1[k][c] = xj1[k]**(i-1) * yj1[k]**(j-1)
+            c = c+1
+
+coeffj1 = np.dot(np.linalg.pinv(coj1),wavj1); # the model
+
+
 fits_o = open('k-order1.txt','w');
-lambda1 = np.zeros((558 ,2048))
+lambda1 = np.zeros((1024 ,2048))
 count  = 0
 # used to predict some value from the training set
-for m in range(466 , 1024):
+for m in range(1 , 1024):
     if m >816 and m<1024:
        for n in range(2047):
            num = 0
            for l in  range(3):
                for g in  range(3):
               
-                   lambda1[m-466][n] = lambda1[m-466][n] + coeff[num]*(n+1)**(l-1)*m**(g-1)
+                   lambda1[m][n] = lambda1[m][n] + coeff[num]*(n+1)**(l-1)*m**(g-1)
                    num = num + 1
-           lambda1[m-466][n] = lambda1[m-466][n]*order_data[m][n]    
+           lambda1[m][n] = lambda1[m][n]*order_data[m][n]    
     else:
 	 if m < 816 and m > 649:
```

Comment:
Like above all sorts of hard coded numbers here. Lets parameterize them and put parameters at top of code.